source/documentBase.py

```diff
@@ -1,14 +1,44 @@
-#A part of NonVisual Desktop Access (NVDA)
-#Copyright (C) 2017 NV Access Limited
-#This file is covered by the GNU General Public License.
-#See the file COPYING for more details.
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2017-2022 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
 
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Optional, Tuple, Union
 from baseObject import AutoPropertyObject, ScriptableObject
 import config
 import textInfos
 import controlTypes
 
 
+_TableID = Union[int, Tuple, Any]
+"""

```

Comment:
I have never seen this pattern of  an unnamed string following an assignment statement. Is it supposed to become the symbol's docstring? I'm not sure that actually works?

Reply:
[epydoc](http://epydoc.sourceforge.net/manual-docstring.html#variable-docstrings) supports this, however I have discovered it is bad practice, and part of the rejected PEP 224.

It appears the correct pattern for this in future is [typing.Annotated](https://docs.python.org/3/library/typing.html#typing.Annotated) in python 3.9.

Reply:
We have been using this pattern already in some contexts such as `BACK_COMPAT_TO`, `appModuleHandler.helperLocalBindingHandle`.